addons/payment/models/payment_token.py

```diff
@@ -159,17 +159,18 @@ def _build_display_name(self, *args, max_length=34, should_pad=True, **kwargs):
         self.ensure_one()
 
         padding_length = max_length - len(self.payment_details or '')
+        if not self.payment_details and not self.create_date:
+            return False
```

Comment:
No need to be fancy in this case; it will never make sense to manually create a payment token from the backend, so let's simply add a small check for the test and that's it.
```suggestion
        if not self.id:
            return ''
            
        padding_length = max_length - len(self.payment_details or '')
```